RoddyCore/test/src/de/dkfz/roddy/RoddyTestSpec.groovy

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2018 German Cancer Research Center (Deutsches Krebsforschungszentrum, DKFZ).
+ *
+ * Distributed under the MIT License (license terms are at https://www.github.com/eilslabs/Roddy/LICENSE.txt).
+ */
+package de.dkfz.roddy
+
+import de.dkfz.roddy.core.ContextResource
+import de.dkfz.roddy.core.ExecutionContext
+import de.dkfz.roddy.execution.io.ExecutionService
+import de.dkfz.roddy.execution.io.LocalExecutionService
+import de.dkfz.roddy.execution.io.fs.FileSystemAccessProvider
+import org.junit.ClassRule
+import spock.lang.Specification
+
+/**
+ * Base class for Spock tests which does some basic initialization stuff.
+ */
+class RoddyTestSpec extends Specification {
+
+    @ClassRule
+    static final ContextResource contextResource = new ContextResource() {
```

Comment:
You said this resource would be available as a single instance and should not get modified -- in particular in parallelized tests, Did you check whether existing tests already modify this resource? Would it be possible to make this field non-final to allow to substitute the value by a copy? Furthermore this will force us to run parallel execution tests of tests (!). Can we not set the value in the subclass's specification and only provide the otherwise duplicated code here? E.g. make this here a trait to mix in with Specifications?

Reply:
I thought about it and I think it is better to just create a context when necessary. This way, we avoid issues when it comes to parallelization.

Reply:
What effect does this decision have on the code? This original code creating a ContextResource in the superclass is still in.